src/Repository/ConsumerRepository.php

```diff
@@ -3,48 +3,30 @@
 namespace App\Repository;
 
 use App\Entity\Consumer;
-use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
-use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\EntityRepository;
+use Symfony\Bridge\Doctrine\Security\User\UserLoaderInterface;
 
 /**
  * @method Consumer|null find($id, $lockMode = null, $lockVersion = null)
  * @method Consumer|null findOneBy(array $criteria, array $orderBy = null)
  * @method Consumer[]    findAll()
  * @method Consumer[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
  */
-class ConsumerRepository extends ServiceEntityRepository
-{
-    public function __construct(ManagerRegistry $registry)
-    {
-        parent::__construct($registry, Consumer::class);
-    }
 
-    // /**
-    //  * @return Consumer[] Returns an array of Consumer objects
-    //  */
-    /*
-    public function findByExampleField($value)
-    {
-        return $this->createQueryBuilder('c')
-            ->andWhere('c.exampleField = :val')
-            ->setParameter('val', $value)
-            ->orderBy('c.id', 'ASC')
-            ->setMaxResults(10)
-            ->getQuery()
-            ->getResult()
-        ;
-    }
-    */
+class ConsumerRepository extends EntityRepository implements UserLoaderInterface
+{
+    // ...
 
-    /*
-    public function findOneBySomeField($value): ?Consumer
+    public function loadUserByUsername($usernameOrEmail)
     {
-        return $this->createQueryBuilder('c')
-            ->andWhere('c.exampleField = :val')
-            ->setParameter('val', $value)
+        return $this->createQuery(
+            'SELECT u
+                FROM App\Entity\Consumer u
+                WHERE u.username = :query
+                OR u.email = :query'
+        )
```

Comment:
Would be nice to use queryBuilder, instead of raw query, not sure about best practises, but so far I think we tried to write with `createQueryBuilder`.

Reply:
I agree, otherwise, no point to use Doctrine

Reply:
ok